clusterloader2/pkg/test/test.go

```diff
@@ -57,9 +58,23 @@ func RunTest(clusterFramework, prometheusFramework *framework.Framework, cluster
 	}
 	ctx := CreateContext(clusterLoaderConfig, clusterFramework, prometheusFramework, state.NewState(), mapping)
 	testConfigFilename := filepath.Base(clusterLoaderConfig.TestScenario.ConfigPath)
-	testConfig, err := ctx.GetTemplateProvider().TemplateToConfig(testConfigFilename, mapping)
+
+	// Set the default values for configuration.
+	var testConfig api.Config
+	testConfig.SetDefaults()
```

Comment:
I'm not a fan of this pattern. You should leave what was here as it was and simply do:
```
testConfig.SetDetauls()
if err := testConfig.Validate(); err != nil {
  // ....
}
```

Reply:
The implementation logic is 

1. set defaults (1 for namespaces,   delete ns flags from command line defaults or settings
2. decode yaml test config
3.  verify the validness of the values

So testConfig needs to be defined  first.

Reply:
As mentioned in: https://github.com/kubernetes/perf-tests/pull/1173#discussion_r410413673
I don't think what you did is what we really want. We don't want to come up with different patterns for things that were already solved in Kubernetes itself.

Reply:
changed as suggested. Please review. Thanks.